generators/example/src/main/scala/ConfigMixins.scala

```diff
@@ -168,3 +176,41 @@ class WithGPIOBoomTop extends Config((site, here, up) => {
     top
   }
 })
+
+// ------------------
+// Multi-RoCC Support
+// ------------------
+
+/**
+ * Map from a hartId to a particular RoCC accelerator
+ */
+case object MultiRoCCKey extends Field[Map[Int, Seq[Parameters => LazyRoCC]]](Map.empty[Int, Seq[Parameters => LazyRoCC]])
+
+/**
+ * Mixin to enable different RoCCs based on the hartId
+ */
+class WithMultiRoCC extends Config((site, here, up) => {
+  case BuildRoCC => site(MultiRoCCKey).getOrElse(site(TileKey).hartId, Nil)
+})
+
+/**
+ * Mixin to add Hwachas to cores based on hart
+ *
+ * For ex:
+ *   Core 0, 1, 2, 3 have been defined earlier
+ *     with hartIds of 0, 1, 2, 3 respectively
+ *   And you call WithMultiRoCCHwacha(Seq(0,1))
+ *   Then Core 0 and 1 will get a Hwacha
+ *
+ * @param harts Seq of harts to specify which will get a Hwacha
+ */
+class WithMultiRoCCHwacha(harts: Seq[Int]) extends Config((site, here, up) => {
```

Comment:
Depending on the intended usage, a varargs parameter might be cleaner syntax:
```scala
// Example: WithMultiRoCCHwacha(0, 1)
class WithMultiRoCCHwacha(harts: Int*)
```

Reply:
Good idea.